File: bone-business/tpa-saas/bone-lowcode/bone-lowcode-infra/src/main/java/com/bone/lowcode/infra/application/vo/page/pageJson/SubmitRule.java
package com.bone.lowcode.infra.application.vo.page.pageJson;

import lombok.Data;

import java.util.List;

@Data
public class SubmitRule {

    private String id;

    /**
     * 字段列表
     */
    private List<SubmitRuleField> fieldList;

    /**
     * 函数类型
     */
    private String functionType;

    /**
     * 函数名
     */
    private String functionName;

    /**
     * 操作符
     */
    private String operator;

    /**
     * 值类型
     */
    private String valueType;

    /**
     * 固定值文本或动态字段id
     */
    private Object value;

    /**
     * 校验方式，0：强校验，阻止作业流程，1：仅提示，可跳过继续作业
     */
    private Byte verifyType;

    /**
     * 提交时不符规则的错误提示文案
     */
    private String errorPrompt;
}

File: bone-business/tpa-sass-react/src/api/rule/submitRule.ts
import { FunctionTypeEnum } from "@/enums/rule/FunctionTypeEnum";
import { FunctionEnum } from "@/enums/rule/FunctionEnum";
import request from "@/utils/request";
import { RuleStatusEnum } from "@/enums/rule/RuleStatusEnum";
import { RuleVerifyTypeEnum } from "@/enums/rule/RuleVerifyTypeEnum";

const BASE_URL = "/page-api/cfg/submitRule";

class SubmitRuleAPI {
  static createSubmitRule(data: SubmitRule) {
    return request.post(`${BASE_URL}/create`, data);
  }
  static getSubmitRuleByPage(pageCode: string, bizIdentityCode: string = "") {
    return request.get(`${BASE_URL}/getRuleByPage`, {
      params: {
        pageCode,
        bizIdentityCode,
      },
    });
  }
  static updateSubmitRule(data: SubmitRule) {
    return request.post(`${BASE_URL}/update`, data);
  }
}

export default SubmitRuleAPI;

export interface SubmitRule {
  id: string;
  fieldIdList: any[];
  functionType: FunctionTypeEnum;
  function: FunctionEnum;
  operator: string;
  valueType: string;
  value: string;
  status: RuleStatusEnum;
  errorPrompt: string;
  pageCode: string;
  bizIdentityCode: string;
  verifyType: RuleVerifyTypeEnum;
}
